tags/splits/splits.py

```diff
@@ -1,52 +1,216 @@
+
+    def split_clone_up():
+        """Clone the active view into a new split opened the up"""
+
+    def split_clone_vertically():
+        """Clone the active view vertically. The left or right orientation is application dependent"""
+
+    def split_clone_horizontally():
+        """Clone the active view horizontally. The top or bottom orientation is application dependent"""
+
+    def split_create():
+        """Creates a new empty split. The orientation is application dependent"""
+
+    def split_clone():
+        """Clones the active view into a new split. The orientation is application dependent"""
+
+    def split_reopen_last():
+        """Reopen the most recently closed split at the same orientation"""
+
+    # Destruction
+    def split_close():
+        """Closes the current split"""
+
+    def split_close_all():
+        """Closes all splits"""
+
+    # Navigation
+    def split_focus_right():
+        """Focus on the split to the right of the current view"""
+
+    def split_focus_left():
+        """Focus on the split to the left of the current view"""
+
+    def split_focus_down():
+        """Focus on the split below the current view"""
+
+    def split_focus_up():
+        """Focus on the split above the current view"""
+
+    def split_focus_next():
+        """Focuses on the next available split"""
+
+    def split_focus_previous():
+        """Focuses on the previous available split"""
+
+    def split_focus_first():
+        """Focuses on the first split"""
+
+    def split_focus_final():
+        """Focuses on the final split"""
+
+    def split_focus_most_recent():
+        """Focuses on the most recently used split"""
+
+    def split_focus_index(index: int):
+        """Focuses on the split at the specified index"""
+
+    def split_focus_negative_index(index: int):
+        """Focuses on the split at the specified index, indexed from the end"""
+
+    # Arrangement
+    def split_move_right():
+        """Move the active split to the right. This creating a new split is application dependent."""
+
+    def split_move_left():
+        """Move the active split to the left. This creating a new split is application dependent."""
+
+    def split_move_down():
+        """Move the active split down. This creating a new split is application dependent."""
+
+    def split_move_up():
+        """Move the active split up. This creating a new split is application dependent."""
+
+    def split_toggle_zen():
+        """Centers the active split (eg: zen mode)"""
+
+    def split_rotate_right():
+        """Rotates the splits to the right"""
+
+    def split_rotate_left():
+        """Rotates the splits to the left"""
+
+    # Resizing
+    def split_toggle_orientation():
+        """Flips the orientation of the active split"""
+
+    def split_toggle_maximize():
+        """Maximizes the active split"""
+
+    def split_layout_reset():
+        """Resets all the split sizes"""
+
+    def split_expand():
```

Comment:
we could have the base definition just call split_expand_width() and split_expand_height()